src/metabase/models/setting.clj

```diff
@@ -313,6 +314,18 @@
   set to true when settings are being written directly via /api/setting endpoints."
   false)
 
+(defn- has-advanced-setting-access?
+  "If `advanced-permissions` is enabled, check if current user has permissions to edit `setting`.
+  Return `false` when `advanced-permissions` is disabled."
+  []
+  (u/ignore-exceptions
+   (classloader/require 'metabase-enterprise.advanced-permissions.common
+                        'metabase.public-settings.premium-features))
+  (if (and ((resolve 'metabase.public-settings.premium-features/enable-advanced-permissions?))
```

Comment:
could `current-user-has-general-permissions?` also handle the feature flag check? then you wouldn't have to worry about resolving the namespace here

maybe it can have a map of defaults values for each setting type to return if the feature flag check returns `false`

Reply:
`current-user-has-general-permissions?` is in `advanced-permissions` namespace so I think it is better to check from where we use it. 
Otherwise we'd need to have that check in many functions in that namespace.